### DISCUSSION TITLE
How to use the preloadData()  trick from Rich Harris sveltesnaps?

### DISCUSSION THREAD

**Discussion by danielres** [sveltejs/kit#10897] [General]
In this video, Rich Harris demonstrates a wonderful trick to leverage preloadData in order to load the content of a different page into a variable.

In this case, it allows, in combination with pushState(), to display the content of the target page in a dialog, without actually leaving the current page. This is a pattern that has been popularized by Instagram, afaik.

The video (at 17:05): https://youtu.be/HdkJTOTY-Js?si=IF-vuHrZxXuXScLx&t=1025

The source code: https://github.com/Rich-Harris/sveltesnaps/blob/fa50e79c3c9d8f10e532f3a751d48ed55336b48e/src/lib/components/PhotoList.svelte#L67

Rich mentions that this was not yet available in the current version of Sveltekit at that time, but would be coming in a future version, as far as I understand.

`pnpm-lock.yaml` from sveltesnaps shows the following:

```
  '@sveltejs/kit':
    specifier: ^1.16.2
    version: 1.16.2(svelte@3.58.0)(vite@4.3.4) 
```    

Is there any plan to make this available?
Does anyone know some nice workaround?

Thanks

**Comment by teemingc** [comment]
See https://github.com/sveltejs/kit/pull/9847 . It's still a work in progress.

**Comment by danielres** [comment]
Great to know!
Thank you for the info :+1:
